less_15.js

```diff
@@ -0,0 +1,195 @@
+// less 15 
+// #1  Quadratic equation
+// Create a function that solves quadratic equation
+// Use discriminant fot finding equation roots
+// Return value of function should be obj {x1, x2}
+// If there are no roots return {x1:null, x2:null}
+// const roots = calculate(1, 5, 6);
+  // {
+  //   x1: -2,
+  //   x2: -3,
+  // };
+
+const calculateQuadratic = (a, b, c) => {
+    const discr = b ** 2 - 4 * a * c;
+    let x1, x2, x ;
+
+    if ( discr > 0 ) {
+        x1 = (-b - Math.sqrt(discr)) / (2 * a);
+        x1 = (Math.round(x1 * 100)) / 100;
+        x2 = (-b + Math.sqrt(discr)) / (2 * a);
+        x2 = (Math.round(x2 * 100)) / 100;
+
+        console.log(x1 ,x2);
+        return x1, x2;
+
+    } else if ( discr === 0 ) {
+        x = - b / (2 * a);
+
+        console.log(x);
+        return x;
+    } else if ( discr < 0 ) {
+        console.log(x1 = null, x2 = null);
+        return x1 = null, x2 = null;
+    }
+
+  };
+  calculateQuadratic(1, 5, 6);
+  calculateQuadratic(1, 5, 2);
+  calculateQuadratic(23, 15, 8);
+  calculateQuadratic(4, 54, 68);
+  calculateQuadratic(88, 57, 5);
+
+//   ===================================================
+
+// #2  Factorial
+// Create a function that finds a factorial of a number
+// const result = factorial(5);
+// 120
+
+
+
+const factorial = (num) => {
+    let result = 1;
+    for (let i = 1 ; i <= num; i++ ) {
+        result = result * i;
+    }
+    console.log(result);
+    return result;
+  };
+  
+factorial(5);
+factorial(7);
+factorial(4);
+
+//   ======================================================
+
+// #3 Calculator
+// Create a function that accepts a callback
+// and infinite number of arguments
+// Callbacks can be: addition, subtraction, multiplication, division
+// const result = calculate(addition, 1, 1, 1, 1);
+//   // 1 + 1 + 1 + 1 = 4
+//   const result = calculate(subtraction, 20, 10, 5);
+//   // 20 - 10 - 5 = 5
+//   const result = calculate(multiplication, 2, 10, 2);
+//   // 2 * 10 * 2 = 40
+//   const result = calculate(division, 20, 10, 2);
+//   // 20 / 10 / 2 = 1
+
+
+const addition = () => '+';
+const subtraction = () => '-';
+const multiplication = () => '*';
+const division = () => '/';
+
+const calculate = (callback, ...rest) => {
+   
+    if ( callback == addition) {
```

Comment:
use strict === operator
Also an interesting solution but not practical
try to extend logic of your callback